Components/Bullet/include/OgreBullet.h

```diff
@@ -81,6 +85,19 @@ _OgreBulletExport btBvhTriangleMeshShape* createTrimeshCollider(const Entity* en
 _OgreBulletExport btConvexHullShape* createConvexHullCollider(const Entity* ent);
 /// create compound shape
 _OgreBulletExport btCompoundShape* createCompoundShape();
+#ifdef OGRE_BUILD_COMPONENT_TERRAIN
+/// height field data
+struct HeightFieldData {
+    /** the position for a center of the shape, i.e. where to place btRigidBody
+     *  or a child of btCompoundShape */
+    Vector3 bodyPosition;
+    /** a heightfield pointer to be freed when
+     * btHeightfieldTerrainShape is freed */
+    float *terrainHeights;
```

Comment:
you could use std::unique_ptr to enforce deletion

Reply:
The problem is I found no way to automagically delete colliders without user-side code help. Would be nice to have some event for that. For now I just create node and store collider in there, however it is not user accessible currently... Need to solve architectural issue here... Would be nice to have some terrain metadata or something to keep shared_ptr there...